### PR TITLE
Store `AssessmentResult` in persistence layer

### PR DESCRIPTION
**Pull request by oxisto**
This PR stores the `AssessmentResult` objects in the orchestrator into the persistent storage. It also fixes remaining authorisation checks in the assessment storage / retrieval layer.

The assessment results in the assessment services are intentionally left in-memory. We might actually remove the "storage" of them altogether. They are basically only used as a debug feature in case you want to look up results that somehow do not end up in the orchestrator. Should we remove them? @anatheka

### CODE REVIEW

File: service/assessment/assessment.go [hunk 1]
```diff
@@ -96,7 +96,9 @@ type Service struct {
 	// hookMutex is used for (un)locking result hook calls
 	hookMutex sync.RWMutex
 
-	// Currently, results are just stored as a map (=in-memory). In the future, we will use a DB.
+	// results will contain all the assessment results of the current run. They will NOT be persisted to the storage,
```
Comment (anatheka) [inline service/assessment/assessment.go:99]:
```suggestion
	// Results will contain all the assessment results of the current run. They will NOT be persisted to the storage,
```

**Comment by anatheka** [comment]
> This PR stores the `AssessmentResult` objects in the orchestrator into the persistent storage. It also fixes remaining authorisation checks in the assessment storage / retrieval layer.
> 
> The assessment results in the assessment services are intentionally left in-memory. We might actually remove the "storage" of them altogether. They are basically only used as a debug feature in case you want to look up results that somehow do not end up in the orchestrator. Should we remove them? @anatheka

I think it is still a general problem, how to make sure that all assessment results end up in the DB. Maybe until we have solved this, we should leave the results stored in memory.

**Review by anatheka — APPROVED** [review]
LGTM after fixing the conflict.

- Shouldn't  we move the assessment_results.go to service/assessment/?

File: service/assessment/assessment.go (hunk 1 above)
Comment (anatheka) [reply service/assessment/assessment.go:99]: What I forgot: I think you have to update the engine as well.

Comment (oxisto) [reply service/assessment/assessment.go:99]: results is unexported -> lower case

**Pull request merged**